src/libgrate/grate-texture.c

```diff
@@ -184,3 +268,216 @@ void grate_texture_clear(struct grate *grate, struct grate_texture *tex,
 	if (err < 0)
 		grate_error("host1x_gr2d_clear() failed: %d\n", err);
 }
+
+static unsigned log2_size(unsigned size)
```

Comment:
We already have a function in grate-3d.c called `log2_size` that has slightly different semantics. I find this a bit confusing, perhaps we could use a different name for one of the, or use this version for both and check `log2_width == pixbuf->width && log2_height == pixbug->height` instead of `log2_width >= 0 && log2_height >= 0` in `grate_3d_set_texture_desc`?